eth/downloader/downloader.go

```diff
@@ -280,6 +290,12 @@ func (d *Downloader) fetchHashes(p *peer, h common.Hash) error {
 				glog.V(logger.Debug).Infof("Peer (%s) responded with empty hash set\n", active.id)
 				return errEmptyHashSet
 			}
+			for _, hash := range hashPack.hashes {
```

Comment:
Suggestion for later:

Turn `hashPack.hashes` in to a `set.Set` and do a `if set.Intersection(hashPack.hashes, badHashes).Size() > 0`